README.md

```diff
@@ -159,7 +159,7 @@ Maps and objects are created like this:
 
 ## How to change source and output folder?
 
-You can specify at spoon its input and output directories with, respectively, `srcFolder` and `outFolder` tags.
+You can specify at spoon its input and output directories with, respectively, `srcFolder`, `testFolder` and `outFolder` tags.
```

Comment:
I'm note sure this new XML tag is actually supported (It's not in the new pom.xml)

Here, we could say that "By default, spoon-maven-plugin considers the source folders (application sources and test sources) configured in the pom.xml"

Reply:
correct, 2 testFolder parameter was part of revision 1 and isn't there anymore. Added your sentence and removed the testFolder tag.

Reply:
Now that we fully rely on the Maven config, I wonder whether we need to keep `srcFolder`. We could directly rely on the Maven `source` configuration. I'm not sure this is for this PR, but I would like to have your opinion on this. WDYT @MartinWitt @vmassol?

Reply:
Yes,TBH I've never understood why the spoon maven plugin was adding this config parameter.  Also it's already possible to add extra src/main or src/test directories using the build helper plugin (https://www.mojohaus.org/build-helper-maven-plugin/index.html) if need be.

Reply:
Note that you still need a config param (boolean) to say whether to run spoon on test sources or not.

Reply:
maybe a bit longer than needed and you can skip to conclusion directy but:
### user
From my point of view, the maven plugin has 3 usergroups: starter, programmer and expert.
Start has zero maven knowledge and some java knowledge.
Programmer knows how to use java and can use maven to some degree.
Expert knows java and maven well, to understand this plugin and future configs easily.

### usecases

We can split a java project in 3 folder groups (groups is here an array of paths).
source: main source code
test: test code containg unit test.
other: maybe some resources, used in unit tests or generated code.
For every folder group, we can easily think of a need for checking some policies with spoon.

Spoon wants to provide an API to let all three user use spoon, with all three folder groups in every combination( e.g check only src and other).
So we want an easy default config but allow configuration to most needs.
A starter user shouldn't need to lookup other maven plugins.

Currently a user needs to define the input folders. As said, maven could do this for the user.
A starter could use the plugin faster, if we add some default setting matching this.
But maybe a user wants to check only one package, so he still needs the input path option.
### Conclusion
To match all user needs, i can currently think of we could add two new booleans, deprecate the srcFolder setting and add inputPath, (*or* keep srcFolder even if it sounds missleading).
`<includeSrc>$Boolean</includeSrc>`
`<includeTest>$Boolean</includeTest>`
`<inputPath>$String</inputPath>`

For the includeSrc/includeTest setting we simply use the maven api and it should be correct in almost all cases. The inputPath setting is the old srcFolder setting. As a default setting i would propose:
`<includeSrc>true</includeSrc>`
`<includeTest>false</includeTest>`
`<inputPath></inputPath>`
  
#### is a boolean flag for includeTest sufficent
No, because what do you do if you want to run analysis on test and not source?

wdyt?